pkg/telemetry/publisher/publisher.go

```diff
@@ -92,13 +92,19 @@ func (p *Publisher) publish(ctx context.Context) error {
 		return fmt.Errorf("mark as published: %s", err)
 	}
 
+	sevenDays := 24 * 7 * time.Hour
+	if err := p.store.DeleteOlderThan(ctx, sevenDays); err != nil {
+		return fmt.Errorf("mark as published: %s", err)
+	}
+
 	return nil
 }
 
 // MetricsStore defines the API for fetching metrics and marking them as published.
 type MetricsStore interface {
-	FetchUnpublishedMetrics(context.Context, int) ([]telemetry.Metric, error)
```

Comment:
This is an API change I made to make it possible to fetch unpublished or published metrics. That was because I needed this kind of flexibility in the tests.